client/network/src/behaviour.rs

```diff
@@ -125,7 +127,11 @@ impl<B: BlockT, H: ExHashT> Behaviour<B, H> {
 		engine_id: ConsensusEngineId,
 		protocol_name: impl Into<Cow<'static, [u8]>>,
 	) {
-		let list = self.substrate.register_notifications_protocol(engine_id, protocol_name);
+		// This is the message that we will send to the remote as part of the initial handshake.
+		// At the moment, we force this to be an encoded `Roles`.
+		let handshake_message = Roles::from(&self.role).encode();
```

Comment:
If you change this message format, what impact does that have?

Reply:
I'm not sure what you mean. We send a single byte here.

Reply:
I was referring to changes of the handshake message format, i.e. instead of the single byte that is sent today, what if we want to extend the handshake message with other information?

Reply:
The intent is that in the future we want to make the handshake message configurable by the code that uses the public API of `sc_network`.
I'm not doing this immediately because figuring out a good API for this is kind of tricky.
I opened #5685

Reply:
OK. Moving from this handshake message to any other (protocol specific) handshake format would require a version change for all protocols that do not decide to stick with just `Roles` as their handshake message, right? And any future handshake message change would likewise require a version change for the respective protocol?

Reply:
Yes! Which is partially why it's hard to figure out a good API for this.